internal/pkg/agent/storage/store/state_store.go

```diff
@@ -43,35 +42,42 @@ type StateStore struct {
 	log   *logger.Logger
 	store storeLoad
 	dirty bool
-	state stateT
+	state state
 
 	mx sync.RWMutex
 }
 
-type stateT struct {
+type state struct {
 	action   action
 	ackToken string
-	queue    []action
+	// TODO: the queue is for scheduled actions. Set its type accordingly.
+	queue []action
 }
 
 // actionSerializer is a combined yml serializer for the ActionPolicyChange and ActionUnenroll
-// it is used to read the yaml file and assign the action to stateT.action as we must provide the
+// it is used to read the yaml file and assign the action to state.action as we must provide the
 // underlying struct that provides the action interface.
+// TODO: get rid of this type
 type actionSerializer struct {
-	ID         string                 `yaml:"action_id"`
-	Type       string                 `yaml:"action_type"`
-	Policy     map[string]interface{} `yaml:"policy,omitempty"`
-	IsDetected *bool                  `yaml:"is_detected,omitempty"`
+	ID         string               `json:"action_id"`
+	Type       string               `json:"action_type"`
+	Data       actionDataSerializer `json:"data,omitempty"`
+	IsDetected *bool                `json:"is_detected,omitempty"`
+	Signed     *fleetapi.Signed     `json:"signed,omitempty"`
+}
+
+type actionDataSerializer struct {
+	Policy map[string]interface{} `json:"policy" yaml:"policy,omitempty"`
```

Comment:
Yes, more things than just policies are delivered in actions. OSqueries, endpoint response actions, etc.

Reply:
@cmacknz this is only for the state store, which only persists `ActionPolicyChange` and `ActionUnenroll`. Therefore there no other possibility for the contents of `data` property rather than `Policy`